ui/src/callWithTelemetryAndErrorHandling.ts

```diff
@@ -3,15 +3,17 @@
  *  Licensed under the MIT License. See License.txt in the project root for license information.
  *--------------------------------------------------------------------------------------------*/
 
-import { OutputChannel, window } from 'vscode';
+import { MessageItem, OutputChannel, window } from 'vscode';
 import TelemetryReporter from 'vscode-extension-telemetry';
 import { IActionContext } from '../index';
 import { IParsedError } from '../index';
+import { DialogResponses } from './DialogResponses';
 import { localize } from './localize';
 import { parseError } from './parseError';
+import { reportIssue } from './reportIssue';
 
 // tslint:disable-next-line:no-any
-export async function callWithTelemetryAndErrorHandling(callbackId: string, telemetryReporter: TelemetryReporter | undefined, outputChannel: OutputChannel | undefined, callback: (this: IActionContext) => any): Promise<any> {
+export async function callWithTelemetryAndErrorHandling(callbackId: string, telemetryReporter: TelemetryReporter | undefined, outputChannel: OutputChannel | undefined, callback: (this: IActionContext) => any, extensionPath?: string): Promise<any> {
```

Comment:
See https://github.com/Microsoft/vscode-azurestorage/blob/32c9f8f1f06c6eb66902eb090b95693582cf1601/src/components/telemetry/reporter.ts#L28

Reply:
Anything in particular I'm looking at? Feels like 6 of one, half dozen of another

Reply:
I.e. you should be able to get extensionPath automatically, right?  Or did I not look at it closely enough?

Reply:
I mean its just a question of if we want to pass in extensionPath or extensionContext. I'm fine with either

Reply:
Sorry, missed that. I don't want to have to pass in either to every call to callWithTelemetry, although it seems like context would be more at hand if we need to do one, right?

What about calling a registration method in the extension's initialization to set this up once?

Reply:
If you don't want to pass stuff in every time, then I would prefer if this method was a part of AzureActionHandler because we already pass extensionContext, outputChannel, and telemetryReporter to its constructor.

That's how we used to have it and I _believe_ you didn't like that because you had to pass `actionHandler` around all the time. But I was trying a [different thing](https://github.com/Microsoft/vscode-azureeventgrid/blob/master/src/extension.ts#L14) in the eventGrid extension where I make global 'extension' variables for things like the outputChannel so that they don't have to be passed around all the time. We could do the same with an instance of the actionHandler. [The reporter](https://github.com/Microsoft/vscode-azurestorage/blob/32c9f8f1f06c6eb66902eb090b95693582cf1601/src/components/telemetry/reporter.ts#L9) that you linked to works in a similar way (aka just a variable that you import rather than passing it around all the time)

Then it would just be the following:
```typescript
import { actionHandler } from './extension'; // we could export actionHandler from a different file if necessary

function example() {
    actionHandler.callWithTelemetryAndErrorHandling('callbackId', () => {
        ...
    });
}
```

Thoughts? Was there some other reason you didn't like having it in AzureActionHandler?

Reply:
Filed an issue to address this separately: https://github.com/Microsoft/vscode-azuretools/issues/137 In terms of this PR, I got rid of most of my changes in index.d.ts so that these changes are purely internal. That means 'reportAnIssue' won't work perfectly if you use callWithTelemetryAndErrorHandling, but I'm okay with that for now.

I agree with your points above, but I just don't want to work on engineering stuff at this point in the release cycle.